### DISCUSSION TITLE
can you tell me how to use python in typescript?

### DISCUSSION THREAD

**Discussion by satya319** [cosinekitty/astronomy#118] [General]
I am new to programming and I am working in my regional calendar. There is python "Swiss" library which have some thing I want to use inside my angular project. Is there anyway to import those library and use inside my angular ionic project? Thank you in advance.

**Comment by cosinekitty** [comment]
Hi @satya319 . Welcome to the project!  I am not sure I understand what you are trying to do.  Are you talking about [this swiss library](https://pypi.org/project/swiss/)?  I can help you if you are asking about Astronomy Engine, but I really don't know much about swiss. Perhaps I am misunderstanding?

**Reply by satya319** [reply]
At First, thank you so much replying me. [pyswisseph](https://pypi.org/project/pyswisseph/) I am talking about this library. and I want to implement [this](https://github.com/webresh/drik-panchanga) in side my Angular project. But this is in python. This "astronomy-engine" have some same features as also this project have. but all features I need All are present inside "drik-panchanga" directly.

**Reply by cosinekitty** [reply]
OK. I do not know very much about Angular programming, but because Angular is based on Node.js, I would imagine it would be difficult to use Python instead of JavaScript directly from an Angular program. If it was absolutely necessary, I suppose one could create a Python program that is executed by an Angular backend as a separate process. In other words, the Angular program could launch the Python program, but they would not directly refer to each other's source code. Then it would be a matter of communicating between the two programs using pipes, files, a REST API, or some other language-independent method.

It might be much more pleasant to either use some Python backend instead of Angular, or to port the astronomy code from Python to TypeScript/JavaScript.  By the way, there is a [TypeScript/JavaScript version of Astronomy Engine](https://github.com/cosinekitty/astronomy/tree/master/source/js), but I don't know whether an equivalent of pyswisseph is available in TS/JS.

**Reply by satya319** [reply]
Thank you for your reply again. I am using [this project](https://github.com/cosinekitty/astronomy) partially inside my project and some are coded by me and some are till inside that " drik-panchang" project. I saw them but those are not usefull for me.
